mmv1/third_party/terraform/provider/provider.go.erb

```diff
@@ -406,6 +381,17 @@ func ProviderConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
 		return nil, diag.FromErr(err)
 	}
 
+	// Verify that universe domains match between credentials and configuration
+	if v, ok := d.GetOk("universe_domain"); ok {
+		if config.UniverseDomain == "" {
```

Comment:
If a user explicitly provides GDU this will throw an error incorrectly, right?

Reply:
it did, updated the logic to check for v.(string) == "googleapis.com"